algosdk/abi/tuple_type.py

```diff
@@ -17,7 +17,7 @@ class TupleType(ABIType):
     """
 
     def __init__(self, arg_types: List[Any]) -> None:
```

Comment:
@algochoi Sharing a few thoughts on 3.7 vs 3.10.  Let me know how you'd like to proceed and/or if you'd like to discuss verbally?

I'm open to a few routes that ultimately lead to 3.10:
* Change the current PR.
* Merge PR as is.  Release.  Wait ~1 week and merge + release follow on PR that bumps to 3.10.

I know the 3.10 move requires some developers to manage Python versions.  Optionally, I am up for adding a brief README or similar tutorial helping developers upgrade.  Example tooling for managing multiple Python versions includes http://asdf-vm.com/ and https://github.com/pyenv/pyenv.

My starting position is that I'd like to avoid a series of incremental upgrades because we're describing bumping to a new minor version.  So, I'd like to minimize the number of jumps made in the iteration.

As maintainers, I'd like to get to more recent Python versions because the language provides new facilities that can help us more easily provide error-free software (e.g. pattern matching).